lib/src/drawConfigContext.dart

```diff
@@ -0,0 +1,282 @@
+      final value = _fetchOrNotSet(param);
+      if (value != null) {
+        switch (param) {
+          case kClientId:
+            clientId = value;
+            break;
+          case kClientSecret:
+            clientSecret = value;
+            break;
+          case kHttpProxy:
+            httpProxy = value;
+            break;
+          case kHttpsProxy:
+            httpsProxy = value;
+            break;
+          case kRedirectUri:
+            _redirectUri = value;
+            break;
+          case kRefreshToken:
+            refreshToken = value;
+            break;
+          case kPassword:
+            password = value;
+            break;
+          case kUserAgent:
+            userAgent ??= value;
+            break;
+          case kUsername:
+            username = value;
+            break;
+          default:
+            throw new DRAWInternalError(
+                'Param $param does not exist in the fieldMap for $type');
+            break;
+        }
+      }
+    } else if (type == kRequiredField) {
+      final value = _fetch(param);
+      if (value != null) {
+        switch (param) {
+          case kOauthUrl:
+            oauthUrl = value;
+            break;
+          case kRedditUrl:
+            redditUrl = value;
+            break;
+          default:
+            throw new DRAWInternalError(
+                'Paramater $param does not exist in the fieldMap for $type');
+            break;
+        }
+      } else {
+        throw new DRAWClientException(
+            'The required field $param, was not found in $kFileName');
+      }
+    }
+  }
+
+  /// Safely return the truth value associated with [item].
+  bool _configBool(final item) {
+    if (item is bool) {
+      return item;
+    } else {
+      final trueValues = ['1', 'yes', 'true', 'on'];
+      return trueValues.contains(item.toLowerCase());
+    }
+  }
+
+  /// Fetch the value under the default site section in the ini file
+  String _fetchDefault(String key) {
+    return _customConfig.get('default', key);
+  }
+
+  /// Fetch value based on the [_primarySiteName] in the ini file.
+  String _fetch(String key) =>
+      (_customConfig.get(_primarySiteName, key) ?? _fetchDefault(key));
+
+  /// Checks if [key] is contained in the parsed ini file, if not returns [kNotSet].
+  ///
+  /// [key] is the key to be searched in the draw.ini file.
+  String _fetchOrNotSet(final key) => (_fetchDefault(key) ?? kNotSet);
+
+  /// Returns path to user level configuration file
+  Uri _getUserConfigPath() {
+    final environment = Platform.environment;
+
+    var osConfigPath;
+
+    /// Load correct config path based on operating system
+    if (Platform.isMacOS) {
+      osConfigPath = Uri.parse(path.join(environment[kMacEnvVar], '.config'));
+    } else if (Platform.isLinux) {
+      osConfigPath = Uri.parse(environment[kLinuxEnvVar]);
+    } else if (Platform.isWindows) {
+      osConfigPath = Uri.parse(environment[kWindowsEnvVar]);
+    }
```

Comment:
Maybe add an `else` here and throw a `DRAWInternalError` if we're not on any of the above OSes (this shouldn't happen, but let's be safe :) ).